scripts/popup.js

```diff
@@ -1,110 +1,210 @@
-function getFormData(){
-    var formElements = $("#metadata-form").children("input:not(#save-page-btn)");
+const extensionDetected = [];
+const SINGLE_FILE_CORE_EXT_ID = 'jemlklgaibiijojffihnhieihhagocma';
+
+function detectExtension(extensionId, callback) {
+  let img;
+  if (extensionDetected[extensionId]) {
+    callback(true);
+  } else {
+    img = new Image();
+    img.src = `chrome-extension://${extensionId}/resources/icon_16.png`;
+    img.onload = function () {
+      extensionDetected[extensionId] = true;
+      callback(true);
+    };
+    img.onerror = function () {
+      extensionDetected[extensionId] = false;
+      callback(false);
+    };
+  }
+}
 
-    var formData = {};
+function getConfig() {
+  return localStorage.config ? JSON.parse(localStorage.config) : {
+    removeFrames: false,
+    removeScripts: true,
+    removeObjects: true,
+    removeHidden: false,
+    removeUnusedCSSRules: false,
+    processInBackground: true,
+    maxFrameSize: 2,
+    displayProcessedPage: false,
+    getContent: true,
+    getRawDoc: false,
+    displayInContextMenu: true,
+    sendToPageArchiver: false,
+    displayBanner: true,
+  };
+}
 
-    formElements.each(function(){
-        formData[ this.name ] = this.value;
+function processable(url) {
+  return ((url.indexOf('http://') === 0 || url.indexOf('https://') === 0));
+}
+
+function invokeSinglePage(tabId, url, processSelection, processFrame) {
+  detectExtension(SINGLE_FILE_CORE_EXT_ID, (detected) => {
+    if (detected) {
+      if (processable(url)) {
+        chrome.extension.sendMessage(SINGLE_FILE_CORE_EXT_ID, {
+          processSelection,
+          processFrame,
+          id: tabId,
+          config: getConfig(),
+        });
+      }
+    } else {
+      alert('missing core');
+    }
+  });
+}
+
+
+chrome.extension.onMessageExternal.addListener((request, sender, sendResponse) => {
+  let blob,
+    url;
+  if (request.processStart) {
+    // singlefile.ui.notifyProcessStart(request.tabId, request.processingPagesCount);
+    if (request.blockingProcess) {
+      chrome.tabs.sendMessage(request.tabId, {
+        processStart: true,
+      });
+    }
+  }
+  if (request.processProgress) {
+    // singlefile.ui.notifyProcessProgress(request.index, request.maxIndex);
+  }
+  if (request.processEnd) {
+    if (request.blockingProcess) {
+      chrome.tabs.sendMessage(request.tabId, {
+        processEnd: true,
+      });
+    }
+    blob = new Blob([(new Uint8Array([0xEF, 0xBB, 0xBF])), request.content], {
+      type: 'text/html',
     });
+    url = URL.createObjectURL(blob);
+    chrome.downloads.download({url:url, filename: "archived-page.html"});
```

Comment:
Let's get the result here and print `archived-page.html` out to the console instead as this is closer to what we ultimately need to do.